x/community/keeper/proposal_handler.go

```diff
@@ -8,10 +8,31 @@ import (
 
 // HandleCommunityPoolLendDepositProposal is a handler for executing a passed community pool lend deposit proposal.
 func HandleCommunityPoolLendDepositProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolLendDepositProposal) error {
+	// move funds from community pool to kavadist so hard position is help by kavadist
+	err := k.distrKeeper.DistributeFromFeePool(ctx, p.Amount, k.moduleAddress)
+	if err != nil {
+		return err
+	}
+	// deposit funds into hard
 	return k.hardKeeper.Deposit(ctx, k.moduleAddress, p.Amount)
 }
 
 // HandleCommunityPoolLendWithdrawProposal is a handler for executing a passed community pool lend withdraw proposal.
 func HandleCommunityPoolLendWithdrawProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolLendWithdrawProposal) error {
-	return k.hardKeeper.Withdraw(ctx, k.moduleAddress, p.Amount)
+	// hard allows attempting to withdraw more funds than there is a deposit for.
+	// this means the amount that gets withdrawn will not necessarily match the amount set in the proposal.
+	// to calculate how much is withdrawn, compare this module's balance before & after withdraw.
+	balanceBefore := k.bankKeeper.GetAllBalances(ctx, k.moduleAddress)
+
+	// withdraw funds from x/hard to kavadist module account
+	err := k.hardKeeper.Withdraw(ctx, k.moduleAddress, p.Amount)
+	if err != nil {
+		return err
+	}
+
+	balanceAfter := k.bankKeeper.GetAllBalances(ctx, k.moduleAddress)
+	totalWithdrawn := balanceAfter.Sub(balanceBefore)
```

Comment:
Nice workaround. For all cases, `balanceAfter > balanceBefore` so this sub is good.

Reply:
yee, technically shouldn't need to worry about balance in x/community since it should be zero, but figured i'd keep the potentially-non-zero-community-balance code for completeness.